src/codeu/chat/client/ClientMessage.java

```diff
@@ -97,18 +98,30 @@ public int currentMessageCount() {
 
   // For m-add command.
   public void addMessage(Uuid author, Uuid conversation, String body) {
+    processAddMessage(author, conversation, body, null);
+    updateMessages(false);
+  }
+
+  public void addFileMessage(Uuid author, Uuid conversation, File file) {
+    processAddMessage(author, conversation, file.getName(), file);
+    updateMessages(false);
+  }
+
+  private void processAddMessage(Uuid author, Uuid conversation, String body, File file){
+
     final boolean validInputs = isValidBody(body) && (author != null) && (conversation != null);
 
-    final Message message = (validInputs) ? controller.newMessage(author, conversation, body) : null;
+    final Message message = (validInputs) ? ((file != null) ?
```

Comment:
I know! Unfortunately (I think) I cannot do a regular if statement because it would involve modifying the final variable message, which is not possible. I'll try to make the indentation look a bit better.